### PR TITLE
Document synthetic source for text and keyword

### PR DESCRIPTION
**Pull request by nik9000**
`text` and `keyword` fields support synthetic _source in a few more configurations now. This documents those configurations.

### CODE REVIEW

**Review by romseygeek — APPROVED** [review]
LGTM

File: docs/reference/mapping/types/text.asciidoc [hunk 1]
```diff
@@ -162,12 +162,14 @@ The following parameters are accepted by `text` fields:
...
-<<copy-to,`copy_to`>>.
-
-Synthetic source always sorts `keyword` fields and removes duplicates, so
-`text` fields are sorted based on the sub-`keyword` field. For example:
-[source,console,id=synthetic-source-text-example]
+a <<keyword-synthetic-source, `keyword`>> sub-field that supports synthetic
+`_source` or if the `text` field sets `store` to `true`. Either way, it may
+not have <<copy-to,`copy_to`>>.
+
+If using a sub-`keyword` field then the values are sorted in the same was as
```
Comment (kilfoyle) [inline docs/reference/mapping/types/text.asciidoc:169]:
```suggestion
If using a sub-`keyword` field then the values are sorted in the same way as
```

File: docs/reference/mapping/types/text.asciidoc [hunk 2]
```diff
@@ -162,12 +162,14 @@ The following parameters are accepted by `text` fields:
...
+a `keyword` field's values are sorted. By default that means sorted with
```
Comment (kilfoyle) [inline docs/reference/mapping/types/text.asciidoc:170]:
```suggestion
a `keyword` field's values are sorted. By default, that means sorted with
```

File: docs/reference/mapping/types/keyword.asciidoc [hunk 3]
```diff
@@ -173,12 +173,11 @@ Dimension fields have the following constraints:
...
 `keyword` fields support <<synthetic-source,synthetic `_source`>> in their
 default configuration. Synthetic `_source` cannot be used together with
-<<ignore-above,`ignore_above`>>, a <<normalizer,`normalizer`>>,
-<<copy-to,`copy_to`>>, or with <<doc-values,`doc_values`>> disabled.
+a <<normalizer,`normalizer`>> or <<copy-to,`copy_to`>>.
 
-Synthetic source always sorts `keyword` fields and removes duplicates. For
-example:
-[source,console,id=synthetic-source-keyword-example]
+By default synthetic source sorts `keyword` fields and removes duplicates.
```
Comment (kilfoyle) [inline docs/reference/mapping/types/keyword.asciidoc:178]:
```suggestion
By default, synthetic source sorts `keyword` fields and removes duplicates.
```

**Review by kilfoyle — APPROVED** [review]
LGTM! 🎸 
Just a couple of super, super small nitpicks.

**Comment by nik9000** [comment]
I tend to omit `,` in lots of places.

**Pull request merged**